econplayground/main/views.py

```diff
@@ -595,65 +597,108 @@ def get_success_url(self):
+        return super(AssignmentListView, self).get(request, *args, **kwargs)
+
+    # def get_object(
+    #       self, queryset: Optional[
+    #           models.query.QuerySet[Any]] = ...) -> models.Model:
+    #     return super().get_object(queryset)
+
+    def post(self, request, pk, *args, **kwargs):
+        print('In here!!!')
+        assignment = Assignment.objects.filter(pk=pk)
+        assignment.update(published=not assignment.published)
+        assignment.save()
+        return super(AssignmentListView, self).get(request, *args, **kwargs)
+
     def dispatch(self, request, *args, **kwargs):
         return super(
             AssignmentListView, self).dispatch(request, *args, **kwargs)
 
     def get_queryset(self):
-        return Assignment.objects.filter(
-            instructor=self.request.user).order_by('created_at')
+        return Assignment.objects.all().order_by('created_at')
+        return Assignment.objects.all().order_by('created_at')
 
-    def get_context_data(self, **kwargs):
-        return super(AssignmentListView, self).get_context_data(**kwargs)
+    def get_context_data(self, *args, **kwargs):
+        ctx = super(AssignmentListView, self).get_context_data(*args, **kwargs)
+        ctx.update({'is_assignment': self.is_assignment})
+        ctx.update({'is_question_bank': self.is_question_bank})
+        ctx.update({'is_question': self.is_question})
+        return ctx
 
 
-class AssignmentDetailView(DetailView):
+class AssignmentDetailView(
+        LoginRequiredMixin, AssignmentInstructorMixin, DetailView):
     model = Assignment
+    is_assignment = True
+    is_question_bank = False
+    is_question = False
 
-    def get_graph_queryset(self):
-        # First, set up graphs based on a users role
-        questions = Question.objects.all()
-
-        # Then apply filtering based on query string params
-        params = self.request.GET
-        if len(params) == 0:
-            return questions.order_by('created_at')
-        return questions.order_by('title')
+    def get_bank_queryset(self):
+        return QuestionBank.objects.all()
 
     def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
+        ctx = super().get_context_data(**kwargs)
+        ctx = super().get_context_data(**kwargs)
 
-        question_list = self.get_graph_queryset()
+        bank_list = self.get_bank_queryset()
+        bank_list = self.get_bank_queryset()
 
-        context['question_list'] = question_list
-        context['all_count'] = question_list.count()
+        ctx.update({'is_assignment': self.is_assignment})
+        ctx.update({'is_question_bank': self.is_question_bank})
+        ctx.update({'is_question': self.is_question})
 
-        return context
+        ctx['bank_list'] = bank_list
+        ctx['all_count'] = bank_list.count()
+        ctx['bank_list'] = bank_list
+        ctx['all_count'] = bank_list.count()
+
+        return ctx
+
+        return ctx
```

Comment:
Duplicate return statement here - this can be removed.

Reply:
Sorry. The rebase got a little messy, thought I caught all of it.

Reply:
No worries, that's what our review process is for after all!